core/shared/src/test/scala-3.x/monocle/focus/FocusCastAsTest.scala

```diff
@@ -0,0 +1,54 @@
+package monocle.focus
+
+import monocle.Focus
+import monocle.Focus._
+
+import scala.annotation.nowarn
+
+
+final class FocusCastAsTest extends munit.FunSuite {
+
+  trait Food { def calories: Int }
+  case class Banana(calories: Int, squishy: Boolean) extends Food
+  case class Apple(calories: Int, color: String) extends Food
+  case class MysteryFood[A](mystery: A, calories: Int) extends Food
+  case class Meal(mainIngredient: Food)
+
+
+  test("Cast a broad thing to a narrow thing, directly on the argument") {
+    val asBanana = Focus[Food](_.as[Banana])
+
+    val foodA: Food = Apple(33, "red")
+    val foodB: Food = Banana(40, true)
+
+    assertEquals(asBanana.getOption(foodB), Some(Banana(40, true)))
+    assertEquals(asBanana.getOption(foodA), None)
+  }
+
+  test("Cast a broad thing to a narrow thing, nested") {
+    val mealAppleColor = Focus[Meal](_.mainIngredient.as[Apple].color)
+
+    val mealA = Meal(Apple(24, "green"))
+    val mealB = Meal(Banana(50, false))
+
+    assertEquals(mealAppleColor.getOption(mealA), Some("green"))
+    assertEquals(mealAppleColor.getOption(mealB), None)
+  }
+
+  test("Cast a narrow thing to a broad thing, which is useless but allowed") {
```

Comment:
should we allow this or request the target to be a subtype of current target?
For example, `GenPrism` is currently defined with:
```scala
object GenPrism {
  def apply[S, A <: S]: Prism[S, A] = macro ...
}
```